Orochi/OrochiUtils.cpp

```diff
@@ -371,6 +371,17 @@ struct OrochiUtilsImpl
 		return 0;
 	}
 
+	static std::string getCacheName( const std::string& path, const std::string& kernelname, std::vector<const char*>* opts ) noexcept
+	{
+		std::string tmp_name = path;
+		for( std::string s : *opts )
+		{
+			if( s.find( "-I" ) != std::string::npos ) continue;
```

Comment:
what about -D?

Reply:
Hi Jam. I'm not sure I understood your question. If we compile kernels with different definitions, for example -DVAL=SOME_NUMBER, then we will generate a cached name - {PATH_TO_SOURCE}+{KERNEL_NAME}+{-DVAL=SOME_NUMBER}.
If we pass another value -DVAL=ANOTHER_NUMBER, then the name will be {PATH_TO_SOURCE}+{KERNEL_NAME}+{-DVAL=ANOTHER_NUMBER} and kernel well be recompiled.
This condition with "-I" just skip path to include.

Reply:
I see. But what if there's a case where we have some "-I" required for "-DVAL=SOME_NUMBER" and not required for "-DVAL=ANOTHER_NUMBER" ? I'm wondering if we should include the whole opts string in this cache file name.

Reply:
I fixed it, checking only the first two characters.
As for including the whole opts line, I implemented it like in the Adl - https://github.com/amdadvtech/adl/blob/5a149e0a6443c55238787cf879058197bee78d91/Adl/HIP/AdlHIP.cpp#L334

Reply:
Hmm. I'm not able to understand why we are skipping -I here. Shouldn't it be included in the cache name too?

Reply:
Yes, you are right. The path will be the same, but the trigger for the new compilation is -D. 
Yes, I tested it.